### PR TITLE
Add button to relaunch the game alongside reconnect when you lose connection

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/66220

## About The Pull Request
.reconnect is not the most reliable thing in the world, and will collapse given any topic call. This will give players a new button in the case that all other buttons fail 

## Testing Photographs and Procedure
image

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl: Mothblocks, RKz
qol: Additional button has been added to tgchat connection closed warning, allowing users to reopen the game in addition to reconnect
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Do we need both buttons, or would just relaunch be good enough? (Will people actually use reconnect if relaunch does the same thing but more powerfully?)

**Comment by Tsar-Salat** [comment]
> (Will people actually use reconnect if relaunch does the same thing but more powerfully?)

I dont understand what you mean by "more powerfully".

It takes longer and might send the client through Byond's ad bullshit or whatever. Its meant to be used when reconnect doesnt work. Reconnect will be still preferable for the average user.

**Comment by PowerfulBacon** [comment]
I feel like people will use relaunch game in situations where reconnect would work, just simply because reconnect has absolutely 0 feedback and no indication that its actually work, where as this one closes and re-opens the game which is a clear indication that it did what you wanted.

**Comment by Tsar-Salat** [comment]
> just simply because reconnect has absolutely 0 feedback and no indication that its actually work

I agree.
However, if someone is pissed off or impatient enough, they are going to X out of the game and reload anyway. The relaunch is just a shortcut to what they would do anyway.

On the reconnect not providing feedback, I agree, however that is outside the scope of this pr.

File: tgui/packages/tgui-panel/reconnect.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+setInterval(() => {
+  Byond.winget('', 'url').then(currentUrl => {
+    // Sometimes, for whatever reason, BYOND will give an IP with a :0 port.
+    if (currentUrl && !currentUrl.match(/:0$/)) {
+      url = currentUrl;
+    }
+  });
+}, 5000);
+
+export const ReconnectButton = (props, context) => {
```
Comment (itsmeow) [inline tgui/packages/tgui-panel/reconnect.tsx]: This should probably be named ReconnectButtons

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui-panel/reconnect.tsx (hunk 1 above)
Comment (Tsar-Salat) [reply tgui/packages/tgui-panel/reconnect.tsx]: done

**Review by PowerfulBacon — APPROVED** [review]

**Review by itsmeow — APPROVED** [review]

**Pull request merged**